src/main/java/org/rulez/demokracia/pdengine/ComputedVote.java

```diff
@@ -2,18 +2,27 @@
 
 import java.io.Serializable;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-public class ComputedVote implements ComputedVoteInterface, Serializable {
+import org.rulez.demokracia.pdengine.dataobjects.Pair;
+
+import jersey.repackaged.com.google.common.collect.Sets;
+import jersey.repackaged.com.google.common.collect.Sets.SetView;
+
+public class ComputedVote implements CanCalculateWinners, ComputedVoteInterface, Serializable {
 
 	private static final long serialVersionUID = 1L;
 	private BeatTable beatTable;
 	private final Vote vote;
 	private BeatTable beatPathTable;
 
+	private transient WinnerCalculator winnerCalculator;
+
 	public ComputedVote(final Vote vote) {
 		this.vote = vote;
+		winnerCalculator = new WinnerCalculatorImpl();
```

Comment:
The fully SOLID compatible solution would be using a WinnerCalculatorFactory with a static getWinnerCalculator method, but I feel that it is over engineering. And in addition It would be hard to mock.
If we would use a Dependency Injection framework (e.g. Spring) it would be this simple:
`
@Autowired
private WinnerCalculator winnerCalculator;
`